rust/emitter/src/ast_emitter.rs

```diff
@@ -374,6 +377,55 @@ impl AstEmitter {
         Ok(())
     }
 
+    fn emit_short_circuit(
+        &mut self,
+        operator: &BinaryOperator,
+        left: &Expression,
+        right: &Expression,
+    ) -> Result<(), EmitError> {
+        self.emit_expression(left)?;
+        let mut jumplist = Vec::new();
+        self.emit_jump(operator, &mut jumplist)?;
+        self.emit.pop();
+        self.emit_expression(right)?;
+        self.emit_jump_target(jumplist);
+        return Ok(());
+    }
+
+    fn emit_jump(
+        &mut self,
+        operator: &BinaryOperator,
+        jumplist: &mut Vec<usize>
+    ) -> Result<(), EmitError> {
+        let offset = 5; // currently missing a check if this is valid
+        let depth = self.emit.bytecode_depth();
+        jumplist.push(depth);
+        match operator {
+            BinaryOperator::Coalesce { .. } => {
+                self.emit.coalesce(offset);
+            }
+            BinaryOperator::LogicalOr { .. } => {
+                self.emit.or(offset);
```

Comment:
What is this `offset` argument here? What does it mean and what is it used for?

Reply:
aha. I believe this is the forward jump offset, described here: https://dxr.mozilla.org/mozilla-central/source/js/src/vm/Opcodes.h#2213

Reply:
it should be just 0 (maybe with some comment), given we replace it later.

Reply:
yes thats true.. i was a bit confused by our current implementation, https://searchfox.org/mozilla-central/source/js/src/frontend/BytecodeEmitter.cpp#238, and I am not sure why we do that. It can also be zero.

Reply:
current impl uses the trailing 4 bytes of forward jump opcodes as a cell of linked list (`JumpList` internal), as an optimization.
if we use separate list, we don't need it.

Reply:
https://searchfox.org/mozilla-central/rev/e076e40ab1290f4e5e67ebd21dc8af753fc05be6/js/src/frontend/JumpList.h#16-54